### PR TITLE
chore: enable eslint

### PR DESCRIPTION
**Pull request by darasus**
Enable ESLint and fix errors

### CODE REVIEW

**Comment by pastelsky** [comment]
Thanks for submitting this change. We should also add this to —
1. Autofix ESLint errors using the current husky precommit hook
2. Add ESLint checks to our CI (`.github/workflows/ci.yml`)
3. Add a lint script to `package.json`
4. Enable ESLint prettier (https://prettier.io/docs/en/integrating-with-linters.html) so formatting changes are linted as well.

**Comment by darasus** [comment]
@pastelsky I think all comments are taken care of. Mind having another look?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
       - run: yarn install
+      - name: Run lint
+        run: yarn next lint
```
Comment (pastelsky) [inline .github/workflows/ci.yml:28]: Why do we have to call "next lint"? What is next providing here additional to what ESLint already provides? 
If its not much, I'd recommend not adding a dependency on `next` for ESLint.

**Review by pastelsky — COMMENTED** [review]
LGTM, except a question about why we're coupling lint with `next`.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": ["next", "prettier"],
```
Comment (pastelsky) [inline .eslintrc.json:2]: We should add `eslint:recommended` as well here (plus a few more plugins). "next", "prettier" are specific rulesets for frameworks.

Happy for this to be done in a follow up PR though

Comment (darasus) [reply .eslintrc.json:2]: I'll do that here in one go

Comment (darasus) [reply .eslintrc.json:2]: Or maybe not, `eslint:recommended` is much more strict, will do that in the following PR

**Review by pastelsky — APPROVED** [review]

**Comment by darasus** [comment]
@pastelsky thanks for approving but i don't have permissions to merge.

**Pull request merged**